src/harness/reference_models/geo/refractivity.py

```diff
@@ -0,0 +1,105 @@
+#    Copyright 2017 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+"""Returns the ITU surface refractivity for a specified lat/lon.
+
+Standalone execution:
+  python refractivity.py lat lng
+"""
+
+import logging
+import math
+import numpy as np
+import os
+import sys
+
+from reference_models.geo import CONFIG
+
+
+class RefractivityIndexer:
+  """RefractivityIndexer class to retrieve refractivity for given world location.
+
+  It retrieves the nearest four refractivity points from ITU data file with
+  1.5 degree grid, and interpolates bilinearly between those points.
+
+  Typical usage:
+    refractor = RefractivityIndexer()
+    r = refractor.Refractivity(19.66, -155.55)
+  """
+  def __init__(self, datafile_or_dir=None):
+    self.ConfigureDataFile(datafile_or_dir, do_load=False)
+    self._num_lat_rows = 121  # Number of latitude rows in the data file
+    self._num_lon_cols = 241  # Number of longitude columns in the data file
+    self._lat_start = 90.0  # Latitude corresponding to first row of file (deg)
+    self._lon_start = 0.0  # Longitude corresponding to first column of file (deg)
```

Comment:
comment from Commscope:

This should be self._lon_start = -180.0 # Longitude corresponding to first column of file (deg). This longitude in the file starts from -180

Reply:
I am sorry, but the spec P452.16 found at https://www.itu.int/rec/R-REC-P.452-16-201507-I/en, clearly specifies that the longitude starts at 0 degree:
"
The maps are contained in the files DN50.TXT and N050.TXT, respectively. The data are from 0d to 360d in longitude and from +90d to –90d in latitude, with a resolution of 1.5d in both latitude and longitude. The data are used in conjunction with the companion data files LAT.TXT and LON.TXT containing respectively the latitudes and longitudes of the corresponding entries (gridpoints) in the files DN50.TXT and N050.TXT
"

Looking at the LON.TXT file, we also see that it starts at 0 degrees.
Can you confirm this? It is the climate file which starts near -180degree.

Reply:
No problem.